subsys/net/lib/nrf_cloud/src/nrf_cloud_fota.c

```diff
@@ -171,10 +173,67 @@ static int fota_settings_set(const char *key, size_t len_rd,
 	return 0;
 }
 
+static int load_fota_settings(void)
+{
+	int ret = settings_subsys_init();
+
+	if (ret) {
+		LOG_ERR("Settings init failed: %d", ret);
+	}
+
+	ret = settings_load_subtree(settings_handler_fota.name);
+	if (ret) {
+		LOG_ERR("Cannot load settings: %d", ret);
+	}
+
+	return ret;
+}
+
+int nrf_cloud_fota_pending_job_validate(enum nrf_cloud_fota_type * const fota_type_out)
+{
+	bool reboot = false;
+	int err = load_fota_settings();
+
+	if (fota_type_out) {
+		*fota_type_out = (err ? NRF_CLOUD_FOTA_TYPE__INVALID : saved_job.type);
+	}
+
+	if (err) {
+		return -EIO;
+	}
+
+	err = nrf_cloud_pending_fota_job_process(&saved_job, &reboot);
+	if (err) {
+		return err;
+	}
+
+	/* Do not enforce a reboot for modem updates since they can also be handled by
+	 * reinitializing the modem lib
+	 */
+	if (saved_job.type != NRF_CLOUD_FOTA_MODEM) {
+		reboot_on_init = reboot;
+	}
+
+	if (save_validate_status(saved_job.id, saved_job.type, saved_job.validate)) {
```

Comment:
if this function fails i don't think there is much that can be done.
the job status will still be updated on nrf cloud, it will just be noted as "not validated".